Event Cards now appears on Modals when Selected

Titan-Ngo opened the pull request:
closes #193 

**Demos:**
Basic Usage:
https://user-images.githubusercontent.com/42155527/116644673-db6be780-a928-11eb-930d-161638c87e92.mp4

How it looks for big titles:
https://user-images.githubusercontent.com/42155527/116644688-e1fa5f00-a928-11eb-8cc1-bb5d568fc035.mp4

**Concerns:**

Logical concern - not sure if relying on the selectedEvent variable from the CalendarPage is good practice:
https://github.com/HKN-UCSD/hkn-server/pull/223/files#diff-abfca6e25270d18b31bae76a4cd58c845041532f0e026cbaaaaf51b3f4c3d060R16

I got an error earlier with Modal and ForwardRef, did this to fix it but not sure if it is the correct way to do so:
https://github.com/HKN-UCSD/hkn-server/pull/223/files#diff-abfca6e25270d18b31bae76a4cd58c845041532f0e026cbaaaaf51b3f4c3d060R18

Accidentally deleted one styles.js file, put it back but now it shows up as a commit.

yuw006 reviewed (approved):
It looks really nice

yuw006 commented:
> It looks really nice

My bad, I thought I got assigned to this one, and I can't delete the approval. Please don't merge yet.

godwinpang commented:
@Titan-Ngo to automagically close an issue, you'll need the close keyword as such

closes #193

godwinpang commented on frontend/src/pages/CalendarPage/components/EventCard/index.js:
```diff
@@ -1,19 +1,25 @@
-      {event && (
+function EventCard({ event, onClose, classes }) {
+  // Listens to whether an event is selected or not from Calendar parent comp
+  const open = !!event;
```
ah - in general components don't control if they appear; it's more customary for the parent to control a child component

Titan-Ngo replied:
> could we not rely on !!s? it's not very clear what the open variable should be.

Honestly, I felt the same way, but for some reason the linting automatically changed it to !!.

Titan-Ngo replied:
Before I just had a check to whether event was null or not.

godwinpang commented:
ty for the screen recordings! looks great :)

Titan-Ngo commented on frontend/src/pages/CalendarPage/components/EventCard/styles.js:
```diff
@@ -6,6 +6,16 @@ const styles = theme => ({
+    justifyContent: 'center',
+    margin: 'auto 10%',
+  },
+  modalDiv: {
```
![image](https://user-images.githubusercontent.com/42155527/118581951-692c3d00-b747-11eb-84f3-3516e2bea4cf.png)
 Changed it to modal-div but got a syntax error, so it will remain as modalDiv.

Titan-Ngo commented:
Recent changes:
- ternary operation to check nullness of `event` instead of !!
- removed `modalDiv` css class. Now using a Grid container instead of a div, but was unable to replace display: flex
- Renamed `open` to `isOpen`
- Imported material UI's Fade in the correct line

thaigillespie commented on frontend/src/pages/CalendarPage/components/EventCard/index.js:
```diff
@@ -10,10 +10,18 @@ import styles from './styles';
+
+  const ref = React.createRef();
+
+  const ModalContent = React.forwardRef(() => (
```
I'm just wondering why you are using createRef() and forwardRef() here but in forwardRef() you don't use the ref being passed down. Any reason as to this?

Otherwise, everything else looks good.

Titan-Ngo replied:
I was getting an error for below that the Modal type needed a reference and an inner forwardRef or something, was not completely sure how to fix it but this did the job. Should I also pass in the ref to ModalContent (forwardRef) too?

thaigillespie replied:
If you're not using it then no need. Yeah if this makes it work then that's fine, but document why you chose to use it w/ a comment or something, and then we can probably merge this in

thaigillespie reviewed (approved):
Overall lgtm, but I do have a question

thaigillespie commented:
@godwinpang you want to take a look since Titan made changes? Since you're one of the reviewers it would be nice if you can follow up further

godwinpang commented on frontend/src/pages/CalendarPage/components/EventCard/index.js:
```diff
@@ -10,10 +10,18 @@ import styles from './styles';
-      {event && (
+function EventCard({ event, onClose, classes }) {
+  // Listens to whether an event is selected or not from Calendar parent comp
+  const isOpen = event ? true : false;
```
can we do event === null instead of the ternary? Javascript Boolean coercion is not that straightforward :p

godwinpang commented on frontend/src/pages/CalendarPage/index.js:
```diff
@@ -201,11 +207,12 @@ class CalendarPage extends React.Component {
             </Grid>
 
             {selectedEvent && (
```
Same here, selectedevent === null

Titan-Ngo replied:
Would 

`!(selectedEvent === null)`

 be a good way to write it?

thaigillespie replied:
selectedEvent !== null works as well

godwinpang reviewed (dismissed):
Lgtm minor nitpicks

Titan-Ngo commented:
- Added comment for forwardRef, as Thai suggested
- Changed syntax of null check to `!== null` in two index.js files

thaigillespie reviewed (approved):
lgtm

thaigillespie commented:
@godwinpang wanna take a look through this to give your review?

thaigillespie commented:
@Titan-Ngo you can start merging this PR in whenever

thaigillespie reviewed (approved):
Lgtm
